api/controllers/UserController.js

```diff
@@ -234,6 +235,318 @@ module.exports = {
     });
   },
 
+  permaBan: function (req, res) {
+    /*  Form parameters:
+          req.params.username: The user who is being banned
+          req.params.banNote: The ban reason to go on the mod log (not visible to banned user, 300 characters max)
+          req.params.banMessage: The note that gets sent with the "you are banned" PM
+          req.params.banlistEntry: The ban reason to appear on the public banlist
+          req.params.duration: The number of days that the user will be banned for.
+        Permaban process:
+          1. Ban user from /r/pokemontrades
+          2. Ban user from /r/SVExchange
+          3. Add "BANNED USER" to user's flair on /r/pokemontrades
+          4. Add user's friend code to /r/pokemontrades AutoModerator config (2 separate lists)
+          5. Add user's friend code to /r/SVExchange AutoModerator config (2 separate lists)
+          6. Remove all of the user's TSV threads on /r/SVExchange
+          7. Add user's info to banlist wiki on /r/pokemontrades
+    */
+    if (!req.user.isMod) {
+      res.json("Not a mod", 403);
```

Comment:
I know that I do this some places, but can we make sure that all res.json we add does it properly.

Eg:

res.json({"error": "Not a mod"}, 403);

I'll start tidying up places where we don't do that already. And feel free to write a function someplace that just takes a string and a number and does this, so we don't have to do it everywhere. It also might be simpler to just return o nthe same line, but that's not necessary.